targets/posix/nanoCLR/targetHAL_Time.cpp

```
@@ -0,0 +1,129 @@
+    ticksAbs %= TIME_CONVERSION__ONEMINUTE;
+    CLR_SafeSprintf(buf, len, "%02d", (int32_t)(ticksAbs / TIME_CONVERSION__ONESECOND));
+
+    if (rest != 0)
+    {
+        CLR_SafeSprintf(buf, len, ".%07d", (uint32_t)rest);
+    }
+
+    return len != 0;
+}
+
+const char *HAL_Time_CurrentDateTimeToString()
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)